demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -215,14 +362,423 @@ static bool getS3ObjectFileSize( size_t * pFileSize,
+                                  response->bodyLen,
+                                  "credentials.sessionToken",
+                                  strlen( "credentials.sessionToken" ),
+                                  ( char ** ) &( sigvCreds->pSecurityToken ),
+                                  &( sigvCreds->securityTokenLen ) );
+
+        if( jsonStatus != JSONSuccess )
+        {
+            LogError( ( "Error parsing sessionToken in the credentials." ) );
+        }
+    }
+
+    if( jsonStatus == JSONSuccess )
+    {
+        /* Get expiration date from HTTP response. */
+        jsonStatus = JSON_Search( ( char * ) response->pBody,
+                                  response->bodyLen,
+                                  "credentials.expiration",
+                                  strlen( "credentials.expiration" ),
+                                  ( char ** ) &( sigvCreds->pExpiration ),
+                                  &( sigvCreds->expirationLen ) );
+
+        if( jsonStatus != JSONSuccess )
+        {
+            LogError( ( "Error parsing expiration date in the credentials." ) );
+        }
+    }
+
+    return jsonStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static void lowercaseHexEncode( const char * pInputStr,
+                                size_t inputStrLen,
+                                char * pHexOutput )
+{
+    static const char digitArr[] = "0123456789abcdef";
+    char * hex = pHexOutput;
+    size_t i = 0U;
+
+    for( i = 0; i < inputStrLen; i++ )
+    {
+        *hex = digitArr[ ( pInputStr[ i ] & 0xF0 ) >> 4 ];
+        hex++;
+        *hex = digitArr[ ( pInputStr[ i ] & 0x0F ) ];
+        hex++;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256( const char * pInput,
+                       size_t ilen,
+                       char * pOutput )
+{
+    return mbedtls_sha256_ret( ( const unsigned char * ) pInput, ilen, ( unsigned char * ) pOutput, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256Init( void * hashContext )
+{
+    mbedtls_sha256_init( ( mbedtls_sha256_context * ) hashContext );
+    return mbedtls_sha256_starts_ret( ( mbedtls_sha256_context * ) hashContext, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256Update( void * hashContext,
+                             const char * pInput,
+                             size_t inputLen )
+{
+    return mbedtls_sha256_update_ret( ( mbedtls_sha256_context * ) hashContext,
+                                      ( const unsigned char * ) pInput,
+                                      inputLen );
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256Final( void * hashContext,
+                            char * pOutput,
+                            size_t outputLen )
+{
+    ( void ) outputLen;
```

Comment:
Please add an assert to check whether the output buffer length is at least 32 bytes long